draft-ietf-quic-recovery.md

```diff
@@ -1000,6 +999,17 @@ The recovery period limits congestion window reduction to once per round trip.
 During recovery, the congestion window remains unchanged irrespective of new
 losses or increases in the ECN-CE counter.
 
+## Loss of protected packets during the handshake
+
+0RTT and 1RTT packets sent prior to handshake completion can arrive before
+the peer has keys to unprotect them.  In those cases, the peer may decide
+not to buffer the packets.  This will cause the packets to never be
+acknowledged and eventually declared lost, despite being delivered to
+the peer.  If the server rejects 0RTT, then the congestion controller
+SHOULD ignore the loss of 0RTT packets.  If any 0RTT or 1RTT packets sent
+prior to knowing the peer has keys to unprotect them are lost, the
+sender's congestion control MAY ignore the loss of those packets if it's
+believe they were received by the peer prior to having the correct keys.
```

Comment:
```suggestion
believed they were received by the peer prior to having the correct keys.
```

Though I might instead say

>A congestion controller MAY choose to ignore loss signals from packets that might have been discarded due to packet protection keys being unavailable.

That's not complete, or a spec you could implement to, but it is less about belief and more about guesswork, which is what this is.

Reply:
What if an attacker deliberately sends random packets at high rate? It is not an unlikely attack. Could this shrink the cnwd to the point where valid communication becomes ineffective? If so, it is not a MAY but a MUST.

Reply:
Traffic injection seems orthogonal to saying that there's a non-congestive form of loss that might get some grace during the handshake.  Injecting enough traffic to overwhelm the link and induce loss can happen at any point in the connection, because it's real loss from real congestion.

Reply:
But the question is if it can be done with far less resources here because it is not only taking bandwidth but also convincing and endpoint to back off. I don't know enough this to really know. It could also be the window is overestimated, which is not such a bad thing in this case.